### PR TITLE
Scripts/Spells: Rasthe - unholy resurrection

### PR DESCRIPTION
**Pull request by ghost**
[//]: # (***************************)
[//]: # (** FILL IN THIS TEMPLATE **)
[//]: # (***************************)

**Changes proposed:**

-  Script effect handle for 180457.
-  Set Health Pct on 25 after cast spell.
-  Value confirmed in youtube video.

**Target branch(es):** 3.3.5/master
- [x] master

**Issues addressed:** Closes #  (insert issue tracker number)

**Tests performed:** (Does it build, tested in-game, etc.)
Analogic behaviour Decimatus Transformation Sickness.

**Known issues and TODO list:** (add/remove lines as needed)

- [ ] 
- [ ] 

[//]: # (**DEPRECATION NOTICE** Instead of creating new PRs to the (old) 6.x branch, make sure the target branch is **master** instead of 6.x.)
[//]: # (**NOTE** Enable the setting "[√] Allow edits from maintainers." when creating your pull request (if not enabled already).)
[//]: # (**NOTE** If this PR __only__ contains SQL files, open a new issue instead and post or link the SQL there.)
[//]: # (**SUGGESTION** When adding new SQL files, name them 9999_99_99_99_db_name.sql to reduce the chance of possible merge conflicts.)

### CODE REVIEW

**Comment by Ulduar** [comment]
Or you can add
``` SQL
-- spell_gen_rasthe_unholy_resurrection
 DELETE FROM `spell_script_names` WHERE `ScriptName` = 'spell_gen_rasthe_unholy_resurrection';
 INSERT INTO `spell_script_names` (`spell_id`,`ScriptName`) VALUES
 (180457,'spell_gen_rasthe_unholy_resurrection');
```

**Comment by Unknown-1000** [comment]
For myself, yes - you can add. But commit for TC command.

**Comment by Ulduar** [comment]
You can use working variant or wait developer answer.

File: src/server/scripts/Spells/spell_generic.cpp [hunk 1]
```diff
@@ -4460,6 +4460,33 @@ class spell_gen_impatient_mind : public SpellScriptLoader
...
+public:
+    spell_gen_rasthe_unholy_resurrection() : SpellScriptLoader("spell_gen_rasthe_unholy_resurrection") { }
+
+    class spell_gen_rasthe_unholy_resurrection_SpellScript : public SpellScript
+    {
+        PrepareSpellScript(spell_gen_rasthe_unholy_resurrection_SpellScript);
+
+        void HandleScript(SpellEffIndex /*effIndex*/)
+        {
+             if (Unit* target = GetHitUnit())
```
Comment (Keader) [inline src/server/scripts/Spells/spell_generic.cpp]: if this spells always hit units ```GetHitUnit()``` cant be null :)

Comment (Ulduar) [inline src/server/scripts/Spells/spell_generic.cpp]: @Keader 
See https://github.com/TrinityCore/TrinityCore/blob/a346fd2baf25bbc3be85f8255f0c5a7dd58beac7/src/server/scripts/Spells/spell_generic.cpp#L4229
Script working? Yes, no need change nothing.

Comment (Keader) [inline src/server/scripts/Spells/spell_generic.cpp]: Sorry, but i disagree.
"Working" is not enough.... it should be implemented more efficiently possible.
If i'm not wrong, ```OnEffectHitTarget``` hook is called when spell hit target... so always GetHitUnit will not be null in this hook when spell only hit units. Can be null if hit objects. 
In this case, he can only make a variable cache, without condition.
like: 
```cpp
Unit* target = GetHitUnit();
target->SetHealth(target->CountPctFromMaxHealth(25));
```

Comment (Ulduar) [inline src/server/scripts/Spells/spell_generic.cpp]: You trying compile after you suggestions - you can get crush after cast.

Comment (Unknown-1000) [inline src/server/scripts/Spells/spell_generic.cpp]: @Ulduar right, spell testing in game and working.
@Keader spell has target 1 (UNIT_CASTER). In this case better add not null for hit object.

**Pull request closed without merging**